site/docs/plugins/commands.md

```diff
@@ -5,10 +5,682 @@ next: false
+example, imagine you have a `login` and a `logout` command. The `login` command
+should only appear for logged out users, and vice versa. This is how you can do
+that with the Commands plugin:
+
+::: code-group
+
+```ts [TypeScript]
+// Use the flavor to create a custom context
+type MyContext = Context & CommandsFlavor;
+
+// Use the new context to instantiate your bot
+const bot = new Bot<MyContext>("token");
+
+// Register the context shortcut
+bot.use(commands());
+
+const loggedOutCommands = new CommandGroup();
+const loggedInCommands = new CommandGroup();
+
+loggedOutCommands.command(
+  "login",
+  "Start your session with the bot",
+  async (ctx) => {
+    await ctx.setMyCommands(loggedInCommands);
+    await ctx.reply("Welcome! Session started!");
+  },
+);
+
+loggedInCommands.command(
+  "logout",
+  "End your session with the bot",
+  async (ctx) => {
+    await ctx.setMyCommands(loggedOutCommands);
+    await ctx.reply("Goodbye :)");
+  },
+);
+
+bot.use(loggedInCommands);
+bot.use(loggedOutCommands);
+
+// By default, users are not logged in,
+// so you can set the logged out commands for everyone
+await loggedOutCommands.setCommands(bot);
+```
+
+```js [JavaScript]
+// Register the context shortcut
+bot.use(commands());
+
+const loggedOutCommands = new CommandGroup();
+const loggedInCommands = new CommandGroup();
+
+loggedOutCommands.command(
+  "login",
+  "Start your session with the bot",
+  async (ctx) => {
+    await ctx.setMyCommands(loggedInCommands);
+    await ctx.reply("Welcome! Session started!");
+  },
+);
+
+loggedInCommands.command(
+  "logout",
+  "End your session with the bot",
+  async (ctx) => {
+    await ctx.setMyCommands(loggedOutCommands);
+    await ctx.reply("Goodbye :)");
+  },
+);
+
+bot.use(loggedInCommands);
+bot.use(loggedOutCommands);
+
+// By default, users are not logged in,
+// so you can set the logged out commands for everyone
+await loggedOutCommands.setCommands(bot);
+```
+
+:::
+
+This way when a user calls `/login`, they'll have their commands list changed to
+contain only the `logout` command. Neat, right?
+
+If you want to prevent, for example, the commands contained in
+`loggedInCommands` from being callable after the user called `/logout`, you must
+implement it in your handlers with your own business logic.
+
+::: danger
+As stated in the
+[Telegram API documentation](https://core.telegram.org/bots/api#botcommand),
+command names can only be form out of:
+
+> 1-32 characters. Can contain only lowercase English letters, digits and
+> underscores.
+
+Therefore calling `setCommands` or `setMyCommands` with anything but
+lower_c4s3_commands will throw an exception. Commands not following this rules
+can still be registered, used and handled, but will never be displayed on the
+user menu as such.
+:::
+
+**Be aware** that `SetCommands` and `SetMyCommands` only affects the commands
+displayed in the user's commands menu, and not the actual access to them. You
```

Comment:
Is this sentence saying the exact same thing as the sentence before the DANGER box? This redundancy feels odd.